test/Maven/CommonSpec.hs

```diff
@@ -0,0 +1,162 @@
+module Maven.CommonSpec (
+  spec,
+) where
+
+import Data.Text (Text)
+import DepTypes (DepType (MavenType), Dependency (..), VerConstraint (CEq))
+
+import Data.Map.Strict qualified as Map
+import Data.Set qualified as Set
+import GraphUtil
+import Graphing qualified
+import Strategy.Maven.Common (MavenDependency (..), filterMavenDependencyByScope)
+import Test.Hspec (Spec, describe, it)
+
+createDepFromScopes :: [Text] -> MavenDependency
+createDepFromScopes scopes = do
+  let dep =
+        Dependency
+          { dependencyType = MavenType
+          , dependencyName = "mygroup:packageOne"
+          , dependencyVersion = Just (CEq "1.0.0")
+          , dependencyLocations = []
+          , dependencyEnvironments = Set.fromList []
+          , dependencyTags = Map.fromList [("scopes", scopes)]
+          }
+  MavenDependency dep (Set.fromList scopes)
+
+spec :: Spec
+spec = do
+  describe "Filter maven deps by scope" $ do
+    it "should not filter from empty include and exclude sets" $ do
+      let compileDep = createDepFromScopes ["compile"]
+      let testDep = createDepFromScopes ["test"]
+      let graph = Graphing.directs [compileDep, testDep] <> Graphing.edges [(compileDep, testDep)]
+      let graph' = filterMavenDependencyByScope (Set.fromList []) (Set.fromList []) graph
+
+      expectDirect [compileDep, testDep] graph'
+      expectDeps [compileDep, testDep] graph'
+      expectEdges [(compileDep, testDep)] graph'
+
+    it "should not filter from empty dep scope and non empty include and exclude sets" $ do
+      -- Dep()
+      -- After filtering with includeSet = Set("compile") and excludeSet = Set("test")
+      -- Dep()
+      let dep = createDepFromScopes []
+      let includeSet = Set.fromList ["compile"]
+      let excludeSet = Set.fromList ["test"]
+      let graph = Graphing.directs [dep]
+      let graph' = filterMavenDependencyByScope includeSet excludeSet graph
+
+      expectDirect [dep] graph'
```

Comment:
should this not be empty, since we specified `scope-only: compile`

Reply:
good point, I think you're right about this